cmd/loom/addressmapper.go

```diff
@@ -56,14 +61,30 @@ func AddIdentityMappingCmd() *cobra.Command {
 				if err != nil {
 					return errors.Wrapf(err, "bad tron private key from file% v", args[1])
 				}
+			case "binance":
+				privkey, err = crypto.LoadECDSA(args[1])
+				if err != nil {
+					return errors.Wrapf(err, "read binance private key from file %v", args[1])
+				}
+				signer := auth.NewBinanceSigner(crypto.FromECDSA(privkey))
+				foreignLocalAddr, err = loom.LocalAddressFromHexString(evmcompat.BitcoinAddress(signer.PublicKey()).Hex())
+				if err != nil {
+					return errors.Wrapf(err, "bad binance private key from file %v", args[1])
+				}
+				sigType = evmcompat.SignatureType_BINANCE
 			}
 
 			foreignAddr := loom.Address{ChainID: chainId, Local: foreignLocalAddr}
-			mapping.To = foreignAddr.MarshalPB()
-			mapping.Signature, err = address_mapper.SignIdentityMapping(user, foreignAddr, privkey)
+			sig, err := address_mapper.SignIdentityMapping(user, foreignAddr, privkey)
 			if err != nil {
 				return errors.Wrapf(err, "sigining mapping with %s key", chainId)
 			}
+			typedSig := append(make([]byte, 0, 66), byte(sigType))
+			// sig is already prefixed with default EIP721 signature type
+			sig = append(typedSig, sig[1:]...)
```

Comment:
Instead of hacking the generated sig get rid of the call to `SignIdentityMapping`, and create the right hash here, then sign it with `GenerateTypedSig`.

Reply:
Fixed